pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -79,6 +79,7 @@ func (s *DiscoveryServer) generateRawClusters(node *model.Proxy, push *model.Pus
 	}
 
 	for _, c := range rawClusters {
+		SetTokenPathForSdsFromProxyMetadata(c, node)
```

Comment:
why introduce the change in this file ? 
cluster.go already passed ```proxy.Metadata``` in [applyTrafficPolicy](https://github.com/istio/istio/blob/release-1.1/pilot/pkg/networking/core/v1alpha3/cluster.go#L646), which is eventually passed to [ConstructSdsSecretConfig](https://github.com/istio/istio/blob/release-1.1/pilot/pkg/model/authentication.go#L127) , and you already handled the customer set token path in that file  ?

Reply:
Without the change in this file, the clusters returned in CDS still use the default path for SDS token, instead of the path in SDS_TOKEN_PATH. Hence the change in this PR, with which the clusters returned correctly use the path in SDS_TOKEN_PATH.

Reply:
that's weird, if true, sounds more like a pilot bug that ```proxy.Metadata``` passed to cluster.go isn't up to date..

Reply:
I would rather fix the root cause - if Metadata is not passed to auth plugin probably other plugins will have same problem.

Reply:
The metada is passed to auth plugin. @costinm Per offline discussion, I will try to add a second cache for SDS and set the SDS token path in the second cache.